added basic working version of TaintTracker

akulpillai opened the pull request:
Modified sandrabeme's code to make a basic working version.
The debug functions simply print instructions that access tainted
memory.

akulpillai committed:
add basic working version of TaintTracker

Modified sandrabeme's code to make a basic working version.
The debug functions simply print instructions that access tainted
memory.

R3x reviewed (changes requested):
Couple of things - addTaint and removeTaint are provided as interfaces to add a tainted memory region and remove tainting a memory region.

So whenever we taint - we use the addTaint and removeTaint functionality. This is important - since this allows us to use multiple tainting options and really easily. Similarly - our taintchecker is just checkTaint() 

Can you rewrite it so that it matches the above. This includes the debug functions as well.

akulpillai committed:
added checkTaint()

taint variables have been moved to the TaintTracker class. The debug
functions now use TaintTracker::checkTaint()
